### PR TITLE
New scenario: Moonlit Night

### PR DESCRIPTION
**Pull request by ampersand55**
#### Summary
Content "New scenario: Moonlit Night"

#### Purpose of change

A new scenario that offers a new type of challenge: to navigate through a group of zombies on a moonlit night where you have the benefit of vision but the monsters don't. 

Also features a game mechanic maybe not all new players know about and teaches the importance of vision and aggro. 

#### Describe the solution

* Set start date to the beginning of a full moon phase.
* Add SCENARIO_SPECIFIC EOC to ensure clear weather and benefit of moonlight. 

~~Not sure why my clear weather EOC isn't working, tried to follow the procedure outlined by [Ramza13](https://github.com/Ramza13) in #63444, which to my understanding was to set conditions with `weather( s )` to where only clear sky is possible and then call `next_weather`.~~

EDIT; Weather EOC fixed by creating a custom weather.

- [x] Fix clear weather EOC 

#### Testing

Works except for the clear weather EOC.

### CODE REVIEW

File: data/json/effects_on_condition/scenario_specific_eocs.json [hunk 1]
```diff
@@ -308,5 +308,11 @@
     "update_mapgen_id": "SCENARIO_SMOKE",
     "method": "json",
     "object": { "place_fields": [ { "field": "fd_smoke", "x": [ 3, 20 ], "y": [ 3, 20 ], "intensity": [ 1, 3 ], "repeat": 60 } ] }
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_scenario_clear_weather",
+    "eoc_type": "SCENARIO_SPECIFIC",
+    "effect": [ { "math": [ "weather('humidity')", "=", "0" ] }, { "math": [ "weather('pressure')", "=", "1050" ] }, "next_weather" ]
```
Comment (GuardianDll) [inline data/json/effects_on_condition/scenario_specific_eocs.json:316]: try to delay next weather by 1 second?

Comment (ampersand55) [reply data/json/effects_on_condition/scenario_specific_eocs.json:316]: Not sure I'm doing this correctly, but I'm only getting the first u_message "EOC_scenario_clear_weather". Also I spawned a hygrometer and checked humidity and it doesn't read 0%.

```json
  {
    "type": "effect_on_condition",
    "id": "EOC_next_weather",
    "effect": [ { "u_message": "EOC_next_weather" }, "next_weather" ]
  },  
  {
    "type": "effect_on_condition",
    "id": "EOC_scenario_clear_weather",
    "eoc_type": "SCENARIO_SPECIFIC",
    "effect": [
      { "math": [ "weather('humidity')", "=", "0" ] },
      { "math": [ "weather('pressure')", "=", "1050" ] },
      { "u_message": "EOC_scenario_clear_weather" },
      { "run_eocs": "EOC_next_weather", "time_in_future": "1 seconds" }
    ]
  }
```

**Comment by ampersand55** [comment]
Ok this was rather annoying to debug as I had to restart until I got specific weather conditions, but I eventually got it working by creating a weather that's guaranteed to be free of fog or rain.

It seems like you can't set `{ "math": [ "weather('humidity')", "=", "0" ] }`, even though it says you can in [NPC.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/NPCs.md). Or at least I couldn't make it to work.

> weather(s) Return or set a weather aspect

Also the issue #63444 can probably be closed as you can copy my methods for creating scenarios with specific weather conditions.

**Comment by IdleSol** [comment]
>It seems like you can't set { "math": [ "weather('humidity')", "=", "0" ] }, even though it says you can in [NPC.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/NPCs.md). Or at least I couldn't make it to work.

It works, it's just that next_weather resets right away

File: data/json/effects_on_condition/scenario_specific_eocs.json (hunk 1 above)
Comment (ampersand55) [reply data/json/effects_on_condition/scenario_specific_eocs.json]: I fixed the issue in another way and I'm resolving this so PR can be merged, but setting things with `{ "math": [ "weather('humidity')", "=", "0" ] },` is still an issue.

**Comment by ampersand55** [comment]
> Bionics weather reader to your rescue

Reading the weather wasn't an issue, it was restarting until you got game where it wasn't foggy. 

I think humidity and other weather stuff isn't global, but defined for each overmap tile, so changing it gets quite messy. But let me know if you find a good solution.

**Comment by IdleSol** [comment]
I haven't dealt with either weather or EoC, so I'm not sure if I can help. But try calling EOC while there is a condition. For example, while the effect is hanging, call EoC.

**Comment by IdleSol** [comment]
If I understand you correctly, you need certain weather for testing? 
https://github.com/CleverRaven/Cataclysm-DDA/blob/9ac7d0512f98decae0b73c842c70d3ef627ced46/data/json/regional_map_settings.json#L1221
```diff
--      "base_wind_season_variation": 50
++      "base_wind_season_variation": 50,
++      "weather_white_list": [ "clear", "cloudy" ],
```
or 
```
      "weather_black_list": [ "sunny" ]
```
https://github.com/CleverRaven/Cataclysm-DDA/blob/9ac7d0512f98decae0b73c842c70d3ef627ced46/doc/REGION_SETTINGS.md?plain=1#L512-L513

**Comment by ampersand55** [comment]
Correct me if I'm wrong, but I think regional settings is only for weathers that occur naturally. As "Moonli"t is a weather that's EOC triggered I don't think it's needed.

**Comment by IdleSol** [comment]
It's not for the mod. It's for testing. As I understand it, you've been loading a lot of times to get fog at game start? Ideally, if you limit the weather to fog only, it would reduce the number of reloads.

**Comment by ampersand55** [comment]
Ah, I misunderstood you. That's a good tip for next time!

**Pull request merged**